src/matrix/room/timeline/Timeline.js

```diff
@@ -236,8 +247,50 @@ export class Timeline {
     /** @package */
     addEntries(newEntries) {
         this._addLocalRelationsToNewRemoteEntries(newEntries);
+        this._updateFetchedEntries(newEntries);
+        this._loadRelatedEvents(newEntries);
         this._remoteEntries.setManySorted(newEntries);
     }
+
+    _updateFetchedEntries(entries) {
+        for (const entry of entries) {
+            const relatedEntry = this._fetchedEventEntries.find(e => e.id === entry.relatedEventId);
```

Comment:
Should _fetchedEventEntries be a `Map` rather? Seems we would only need to look things up by id and never iterate otherwise?